packages/app/src/pages/Runs.vue

```diff
@@ -1,14 +1,89 @@
 <template>
-  <div>
-    <h2>Runs Page</h2>
+  <div class="relative p-24px h-full overflow-y-scroll">
+    <transition
+      v-if="query.fetching.value || activeProject?.cloudProject?.runs?.nodes.length"
+      name="fade"
+    >
+      <RunsSkeleton v-if="query.fetching.value" />
+      <div
+        v-else-if="activeProject?.cloudProject?.runs?.nodes.length"
+        data-cy="runs"
+      >
+        <RunCard
+          v-for="run of activeProject.cloudProject.runs.nodes"
+          :key="run.id"
+          :gql="run"
+        />
+      </div>
+    </transition>
+    <RunsConnect
+      v-else-if="query.data.value && (!activeProject?.projectId || !query.data.value.cloudViewer?.id)"
+      :is-logged-in="!!query.data.value.cloudViewer?.id"
+      :gql="query.data.value"
+    />
+    <RunsEmpty
+      v-else
+      :project-id="activeProject?.projectId || ''"
+      :project-name="activeProject?.title || ''"
+    />
   </div>
 </template>
 
+<script lang="ts" setup>
+import { computed } from 'vue'
+import { gql, useQuery } from '@urql/vue'
+import { RunsDocument } from '../generated/graphql'
+import RunCard from '../runs/RunCard.vue'
+import RunsSkeleton from '../runs/RunsSkeleton.vue'
+import RunsConnect from '../runs/RunsConnect.vue'
+import RunsEmpty from '../runs/RunsEmpty.vue'
+
+gql`
+query Runs {
+  app {
+    activeProject {
+      id
+      title
+      projectId
+      cloudProject {
+        id
+        runs {
```

Comment:
This isn't valid. I don't have a validation on this in the mock schema but this will break if executed against the actual remote GraphQL API. Just keep the `first: 10` for now and we can do pagination later. I will add a validation in for this in the meantime, but prob in a separate PR

Reply:
removed